bittensor/synapse.py

```diff
@@ -335,8 +336,26 @@ def __setattr__(self, name, value):
             raise AttributeError(
                 "body_hash property is read-only and cannot be overridden."
             )
+        if name == "required_hash_fields":
+            raise AttributeError(
+                "required_hash_fields property is read-only and cannot be overridden."
+            )
         super().__setattr__(name, value)
 
+    @property
+    def required_hash_fields(self) -> List[str]:
+        """
+        Get the list of required fields for the Synapse instance.
+
+        This method retrieves the list of required fields for the Synapse instance
+        by looking up the schema definition for the current class and returning the
+        list of required fields.
+
+        Returns:
+            List[str]: A list of required fields for the Synapse instance.
+        """
+        return []
```

Comment:
As a default, perhaps it should just return all the hash fields. The property gives subnet designers flexibility if they want to add additional fields in the future while providing backward compatibility.